app/controllers/nodes_controller.rb

```diff
@@ -89,6 +89,181 @@ class NodesController < AuthenticatedUsersController
+    #     "Node[]": [
+    #       "3a95fd96-876d-439a-a64d-70332c069aaa",
+    #       "3943fab1-9ed2-4eb6-8121-5a69499c4568",
+    #       "b10e00e9-e444-41c2-9517-df2118b42731"
+    #     ],
+    #     "fsid": "140cd3d5-58e4-4935-a954-d946ceff371d",
+    #     "public_network": "192.168.128.0/24",
+    #     "cluster_network": "192.168.220.0/24",
+    #     "ceph_conf_overrides": {
+    #       "global": {
+    #         "osd_pool_default_pg_num": 128,
+    #         "pool_default_pgp_num": 1
+    #       }
+    #     },
+    #     "node_configuration": {
+    #       "3a95fd96-876d-439a-a64d-70332c069aaa": {
+    #         "role": "ceph/mon",
+    #         "provisioning_ip": "10.0.0.24",
+    #         "monitor_interface": "eth0"
+    #       },
+    #       "3943fab1-9ed2-4eb6-8121-5a69499c4568": {
+    #         "role": "ceph/osd",
+    #         "provisioning_ip": "10.0.0.29",
+    #         "journal_size": 5192,
+    #         "journal_colocation": "false",
+    #         "storage_disks": [
+    #           {
+    #             "device": "/dev/sda",
+    #             "journal": "/dev/sdc"
+    #           },
+    #           {
+    #             "device": "/dev/sdb",
+    #             "journal": "/dev/sdc"
+    #           }
+    #         ]
+    #       },
+    #       "b10e00e9-e444-41c2-9517-df2118b42731": {
+    #         "role": "ceph/osd",
+    #         "provisioning_ip": "10.0.0.30",
+    #         "journal_colocation": "true",
+    #         "storage_disks": [
+    #           {
+    #             "device": "/dev/sda"
+    #           },
+    #           {
+    #             "device": "/dev/sdb"
+    #           }
+    #         ]
+    #       }
+    #     }
+    #   }
+    # }
+
+    missing_params = []
+    ['sds_type', 'node_configuration'].each do |param|
+      missing_params << param unless body[param] and not body[param].empty?
+    end
+    halt 401, { errors: { missing: missing_params } } unless missing_params.empty?
+
+    # Check that the list of nodes is valid
+    node_identifier = body['node_identifier']
+
+    halt 401, { errors: { invalid: "'node_identifier', if specified, must be either 'uuid' or 'ip', provided: '#{node_identifier}'." } } \
+      if node_identifier and \
+        not ['uuid','ip'].include? node_identifier
+
+    nodes = {}
+    node_ids = body['node_configuration'].keys
+
+    unavailable_nodes = []
+    node_ids.each do |node_id|
+      node = case node_identifier
+             when 'ip'
+               Tendrl::Node.find_by_ip(node_ip)
+             when 'uuid'
+               Tendrl::Node.new(uuid)
+             end
+
+      if node.nil? or not node.exist?
+        unavailable_nodes << node_id
+        next
+      end
+
+      nodes[node.uuid] = body['node_configuration'][node_id]
+    end
+
+    halt 404, { errors: { missing: "Unavailable nodes: #{unavailable_nodes.join(', ')}." } } unless unavailable_nodes.empty?
+
+    parameters = {}
```

Comment:
What happens to the parameters? The job creation never gets those, you are passing body to create.